spec/forms/hub/take_action_form_spec.rb

```diff
@@ -93,4 +197,162 @@
       end
     end
   end
+  describe "#take_action!" do
+    context "when tax_return_id is not part of clients tax returns" do
+      let(:intake) { create :intake }
+      let(:form) { Hub::TakeActionForm.new(client, current_user, { tax_return_id: create(:tax_return).id }) }
+
+      it "throws an error" do
+        expect { form.take_action! }.to raise_error TakeActionFormError
```

Comment:
I saw this custom exception, and I realized we make custom exceptions rarely, and I think I figured out a pattern by which we can avoid creating the custom exception. Non-blocking suggestion:

One pattern I've seen is to use ActiveModel validators within forms. This way, the controller can use a pattern like

```
if @take_action_form.save
  redirect_to it_worked_path
else
  render :edit
end
```

Then the TakeActionForm's `save` method might look something like:

```
def save
  return false unless valid?
  
   doActualStuff()
end
```

It would require renaming `take_action!` to `save`, so it would be a somewhat sizable change. Curious for your thoughts about that approach. (Non-blocking.)